BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -87,4 +87,36 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
     return self;
 }
 
+- (id)initWithCoder:(NSCoder *)coder
+{
+    self = [super init];
+    if (self) {
+        _type = [coder decodeObjectOfClass:[BOXRepresentation class] forKey:NSStringFromSelector(@selector(type))];
```

Comment:
I think these need to be the classes of the things actually being decoded. So, `NSString.class` in this case.

Reply:
done